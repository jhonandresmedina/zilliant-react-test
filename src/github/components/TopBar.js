import React from 'react';
import { Toolbar, Avatar, Button } from 'react-md';

const TopBar = ({ user, updateAll }) => {
    const avatar = user ? <Avatar key='avt' src={user.get('avatar_url')} /> : <Avatar key='avt' />;
    const name = user ? user.get('login') : '';
    const button = (
        <Button onClick={updateAll} icon>
            replay
        </Button>
    );

    return <Toolbar fixed colored nav={avatar} title={name} actions={button} />;
};

export default TopBar;
