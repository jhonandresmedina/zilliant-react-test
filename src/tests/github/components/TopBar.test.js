//@vendor
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { fromJS } from 'immutable';

//@components
import TopBar from '../../../github/components/TopBar';

Enzyme.configure({ adapter: new Adapter() });

const buildComponent = props => {
    const user = fromJS({
        avatar_url: 'https://example-url/image.png',
        login: 'login',
    });

    return <TopBar user={user} />;
};

describe('TopBar test', () => {
    it('renders correctly', () => {
        const wrapper = shallow(buildComponent());
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
