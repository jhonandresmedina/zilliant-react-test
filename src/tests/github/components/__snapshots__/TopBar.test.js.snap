// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar test renders correctly 1`] = `
<Toolbar
  actions={
    <withInk(withTooltip(Button))
      icon={true}
      inkTransitionEnterTimeout={450}
      inkTransitionLeaveTimeout={300}
      inkTransitionOverlap={150}
    >
      replay
    </withInk(withTooltip(Button))>
  }
  colored={true}
  component="header"
  fixed={true}
  nav={
    <Avatar
      src="https://example-url/image.png"
      suffixes={
        Array [
          "red",
          "pink",
          "purple",
          "deep-purple",
          "indigo",
          "blue",
          "light-blue",
          "cyan",
          "teal",
          "green",
          "light-green",
          "lime",
          "yellow",
          "amber",
          "orange",
          "deep-orange",
          "brown",
          "grey",
          "blue-grey",
        ]
      }
    />
  }
  singleColor={true}
  title="login"
/>
`;
